File: backend/PharmaCore.Core/Entities/Supplier.cs


namespace PharmaCore.Core.Entities
{
    public class Supplier
    {
        public int SupplierId { get; set; }
        public string CompanyName { get; set; } = string.Empty;
        public string ContactPerson { get; set; } = string.Empty;
        public string Phone { get; set; } = string.Empty;
        public string? Email { get; set; }
        public bool IsActive { get; set; }

        public virtual ICollection<PurchaseOrder> PurchaseOrders { get; set; } = new HashSet<PurchaseOrder>();
        public virtual ICollection<StockBatch> StockBatches { get; set; } = new HashSet<StockBatch>();
    }
}

File: frontend/src/types/supplier.ts
export interface Supplier {
  supplierId: number;
  name: string;
  phone: string;
  email?: string | null;
  contactPerson: string;
  isActive: boolean;
}

export interface CreateSupplierPayload {
  supplierName: string;
  phone: string;
  email?: string | null;
  contactPerson: string;
}

export interface UpdateSupplierPayload {
  supplierName: string;
  phone: string;
  email?: string | null;
  contactPerson: string;
  isActive: boolean;
}

export type SupplierFilter = "all" | "active" | "inactive";
